### PR TITLE
Kafka: split SASL and TLS config

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

### CODE REVIEW

File: content/docs/2.0/scalers/apache-kafka.md [hunk 1]
```diff
@@ -34,21 +33,25 @@ triggers:
...
-- `ca` Certificate authority file for TLS client authentication. Optional. If authmode is sasl_ssl, this is required.
-- `cert` Certificate for client authentication. Optional. If authmode is sasl_ssl, this is required.
-- `key` Key for client authentication. Optional. If authmode is sasl_ssl, this is required.
+SASL:
+- `sasl`: Kafka SASL auth mode. Optional. If not set, SASL for Kafka is not used. If set, it must be one of `plaintext`, `scram_sha256` or `scram_sha512`.
+- `username`: Optional. If `sasl` is set, this is required.
+- `password`: Optional. If `sasl` is set, this is required.
+
+TLS:
+- `tls`: Optional. To enable SSL auth for Kafka, set this to `enable`. If not set, TLS for Kafka is not used.
```
Comment (zroubalik) [inline content/docs/2.0/scalers/apache-kafka.md:46]: I am not sure, what is the best way to handle this, whether this apporach, with `tls: enable`, or somehing like `enableTls: true/false` ? I have chosen the current approach to have the config similar to SASL one

Comment (ppatierno) [reply content/docs/2.0/scalers/apache-kafka.md:46]: I like just `tls` :-)

Comment (tomkerkhove) [reply content/docs/2.0/scalers/apache-kafka.md:46]: I think for others we use "use" or "enable".

It you read a spec and it just says "tls", what does it mean? Nothing beats a bool with a user friendly name

Comment (zroubalik) [reply content/docs/2.0/scalers/apache-kafka.md:46]: @tomkerkhove well, TLS is kinda selfexplanatory and as I said I kinda like the consistency between the properties, ie. `sasl`, `tls`

Comment (tomkerkhove) [reply content/docs/2.0/scalers/apache-kafka.md:46]: Fine by me, just wanted to elaborate on my thinking

File: content/docs/2.0/migration.md [hunk 2]
```diff
@@ -87,3 +87,9 @@ spec:
 **Example of v2 ScaledJob**
 
 TODO
+
```
Comment (tomkerkhove) [inline content/docs/2.0/migration.md:90]:
```suggestion
```

File: content/docs/2.0/migration.md [hunk 3]
```diff
@@ -87,3 +87,9 @@ spec:
...
+
+### Scalers
+ 
+**Kafka**
```
Comment (tomkerkhove) [inline content/docs/2.0/migration.md:94]: Nit : Should we use a list?

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**